src/scripts/main.js

```diff
@@ -1,3 +1,294 @@
 'use strict';
 
-// write your code here
+const buttonAction = document.querySelector('.button');
+const messages = document.querySelectorAll('.message');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+const gameScore = document.querySelector('.game-score');
+const fieldLength = 4;
+const up = 'ArrowUp';
+const down = 'ArrowDown';
+const left = 'ArrowLeft';
+const right = 'ArrowRight';
+let winning;
+let score;
+let field;
+
+buttonAction.addEventListener('click', () => {
+  score = 0;
+  winning = false;
+
+  field = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  addNewNumber();
+  addNewNumber();
+
+  allMessagesHidden();
+
+  buttonAction.innerHTML = 'Reset';
+  buttonAction.classList.remove('start');
+  buttonAction.classList.add('restart');
+
+  render();
+});
+
+function addNewNumber() {
+  if (includesZero()) {
+    let row = field[randomNumber(4)];
+    const zeroIndex = row.indexOf(0);
+
+    if (zeroIndex !== -1) {
+      const allIndexZero = row.reduce((acc, el, i) => {
+        if (el === 0) {
+          acc.push(i);
+        }
+
+        return acc;
+      }, []);
+
+      const randomZero = allIndexZero[randomNumber(allIndexZero.length)];
+      const rowIndex = field.indexOf(row);
+
+      field[rowIndex][randomZero] = randomNumber();
+    } else {
+      row = field[randomNumber(4)];
+    }
+  }
+}
+
+function randomNumber(max) {
+  if (!max) {
+    return Math.random() < 0.2 ? 4 : 2;
+  }
+
+  return Math.floor(Math.random() * max);
+}
+
+function render() {
+  const cell = document.querySelectorAll('.field-cell');
+
+  gameScore.innerHTML = score;
+
+  if (winning) {
+    messageWin.classList.remove('hidden');
+  }
+
+  if (!includesZero() && isGameOver()) {
+    messageLose.classList.remove('hidden');
+  }
+
+  for (let i = 0; i < fieldLength; i++) {
+    for (let j = 0; j < fieldLength; j++) {
+      const index = i * fieldLength + j;
+
+      if (field[i][j] !== 0) {
+        cell[index].innerHTML = field[i][j];
+        cell[index].className = `field-cell field-cell--${field[i][j]}`;
+      } else {
+        cell[index].innerHTML = '';
+        cell[index].className = 'field-cell';
+      }
+    }
+  }
+}
+
+document.addEventListener('keyup', (e) => {
+  switch (e.code) {
+    case up:
+      moveUp();
+      break;
+
+    case down:
+      moveDown();
+      break;
+
+    case left:
+      moveLeft();
+      break;
+
+    case right:
+      moveRight();
+      break;
+
+    default:
+      break;
+  }
+});
+
+function moveLeft() {
+  for (let i = 0; i < fieldLength; i++) {
+    let row = filterZero(field[i]);
+
+    addZeros(row);
+
+    summationNumbers(row);
+
+    row = filterZero(row);
+
+    addZeros(row);
+
+    field[i] = row;
+  }
+
+  addNewNumber();
+  render();
```

Comment:
you can move these actions to `keyup `handler to avoid duplication
![image](https://user-images.githubusercontent.com/98764641/225845029-9ddf6712-36cf-4d02-bc71-eb3977ef3b6f.png)